components/org.wso2.carbon.identity.scim2.common/src/main/java/org/wso2/carbon/identity/scim2/common/impl/SCIMRoleManager.java

```diff
@@ -857,15 +865,14 @@ private void doUpdateGroups(String roleId, Set<String> newGroupIDList, Set<Strin
         }
     }
 
-    private List<String> getUserIDList(List<String> userList, String tenantDomain) throws CharonException {
+    private List<String> getUserIDList(List<String> userList, String tenantDomain) throws BadRequestException {
 
         List<String> userIDList = new ArrayList<>();
         for (String user : userList) {
             try {
                 userIDList.add(getUserIDByName(user, tenantDomain));
             } catch (IdentityRoleManagementException e) {
-                throw new CharonException(String.format("Error occurred while getting the user id " +
```

Comment:
Don't we need to handle the server exception that can be happened when trying to resolve ID from userName

Reply:
The only error thrown from that method implementation is with InvalidRequest. Will add a comment.

Reply:
Updated.